zeppelin-web/src/app/notebook/notebook.controller.js

```diff
@@ -606,6 +609,120 @@ function NotebookCtrl($scope, $route, $routeParams, $location, $rootScope,
     websocketMsgSrv.getInterpreterBindings($scope.note.id);
   };
 
+  $scope.toggleSelection = function(paragraphId) {
+    let paragraphs = $scope.selectedParagraphsIds;
+    if (paragraphs.has(paragraphId)) {
+      paragraphs.delete(paragraphId);
+    } else {
+      paragraphs.add(paragraphId);
+    }
+  };
+
+  $scope.clearSelection = function() {
+    if ($scope.selectedParagraphsIds !== null) {
+      $scope.selectedParagraphsIds.clear();
+    }
+  };
+
+  $scope.isSelectionMode = function() {
+    if ($scope.selectedParagraphsIds === null || $scope.selectedParagraphsIds.size === 0) {
```

Comment:
What do you think if multi-paragraph toolbar will popup when at least two paragraphs selected?

Reply:
@Tagar, I do not quite understand what you mean.

Reply:
consider changing this line to 
`if ($scope.selectedParagraphsIds === null || $scope.selectedParagraphsIds.size < 2) {` 
so the toolbar for multi-paragraph actions will show up when at least 2 paragraphs are selected.